__tests__/test.js

```diff
@@ -92,6 +92,55 @@ describe("GET /api/articles ", () => {
   });
 });
 
+describe("POST /api/articles/:article_id/comments ", () => {
+  test(".../:article_id/comments adds new comment to given article_id", () => {
+    const newComment = {
+      username: "icellusedkars",
+      body: "To be, or not to be, that is the question",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(201)
+      .then((comment) => {
+        expect(comment).toHaveProperty(
+          "text",
+          `${newComment.username} commented ${newComment.body}`
+        );
+      });
+  });
+  test("returns not an article when article not in DB", () => {
+    const article_id = 9999;
+    const newComment = {
+      username: "icellusedkars",
+      body: "To be, or not to be, that is the question",
+    };
+    return request(app)
+      .post(`/api/articles/${article_id}/comments`)
+      .send(newComment)
+      .expect(404)
+      .then((err) => {
+        expect(err).toHaveProperty("text", `${article_id} is not an article`);
+      });
+  });
+  test("returns 'not registered user when username not in DB", () => {
+    const article_id = 1;
+    const newComment = {
+      username: "Hamlet",
+      body: "To be, or not to be, that is the question",
+    };
+    return request(app)
+      .post(`/api/articles/${article_id}/comments`)
+      .send(newComment)
+      .expect(404)
+      .then((err) => {
+        expect(err).toHaveProperty(
+          "text",
+          `The username "${newComment.username}" does not exist`
+        );
```

Comment:
Your error testing is very good! Another test you should add to this is that the serve responds appropriately if the vital keys `body` or `author` are missing from the request body